### PR TITLE
JP-3695: Clean up unnecessary copies

### PR DESCRIPTION
**Pull request by penaguerrero**
Partially Resolves [JP-3695](https://jira.stsci.edu/browse/JP-3695)

Closes #

This PR addresses removing the unnecessary copies of all steps in the Detector1 pipeline. The same needs to be done for the stage2 pipelines.

Copies are made at the _step.py level, only if the step is not skipped.

**Checklist for PR authors (skip items if you don't have permissions or they are not applicable)**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [x] updated or added relevant tests
- [x] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] All comments are resolved
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

**Comment by penaguerrero** [comment]
reg tests: https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1656/

**Review by tapastro — CHANGES_REQUESTED** [review]
Some changes requested, and needs a changelog entry. We should be able to get this into the build!

The regression test will hopefully be replaced by benchmarks come the end of next build, but we can keep it in until they are functional.

File: jwst/emicorr/emicorr.py [hunk 1]
```diff
@@ -79,7 +79,6 @@ def do_correction(input_model, emicorr_model, save_onthefly_reffile, **pars):
         will operate.  Valid parameters include:
             save_intermediate_results - saves the output into a file and the
                                         reference file (if created on-the-fly)
-            user_supplied_reffile - reference file supplied by the user
```
[the remark this answers is not in the document]
Comment (penaguerrero) [reply jwst/emicorr/emicorr.py:82]: I removed this line already, not sure why it still showed up. I double checked now and it is not there.

File: jwst/ramp_fitting/ramp_fit_step.py [hunk 2]
```diff
@@ -449,22 +452,19 @@ def process(self, input):
             if self.algorithm == "GLS":
                 buffsize //= 10
 
-            int_times = input_model.int_times
+            int_times = result.int_times
 
-            # Before the ramp_fit() call, copy the input model ("_W" for weighting)
-            # for later reconstruction of the fitting array tuples.
-            input_model_W = copy.copy(input_model)
```
[the remark this answers is not in the document]
Comment (penaguerrero) [reply jwst/ramp_fitting/ramp_fit_step.py:456]: I reverted these lines.

File: jwst/emicorr/emicorr.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (tapastro) [reply jwst/emicorr/emicorr.py:82]: It's still listed in the documentation pages - that's where it also needs to be pulled.

**Comment by penaguerrero** [comment]
> Some changes requested, and needs a changelog entry. 
We had talked about this in the meetings and had decided that no log entry was needed since it did not change the behavior for users. Thinking about it a bit more, I will go ahead and add a log entry, for tracking purposes.

**Comment by tapastro** [comment]
> > Some changes requested, and needs a changelog entry.
>  We had talked about this in the meetings and had decided that no log entry was needed since it did not change the behavior for users. Thinking about it a bit more, I will go ahead and add a log entry, for tracking purposes.

That's fine, you can mark it with the appropriate no-changelog tag then.

**Comment by tapastro** [comment]
Latest regtest run was clean: https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1741/

Merging!

**Pull request merged**